sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -95,7 +95,7 @@ class StringFunctionsSuite extends QueryTest with SharedSQLContext {
   }
 
   test("non-matching optional group") {
-    val df = Seq("aaaac").toDF("s")
+    val df = Seq(Tuple1("aaaac")).toDF("s")
```

Comment:
OK, I guess that Scala 2.11 has an implicit conversion that Scala 2.10 does not that would make this conversion automatically, hence the PR builder succeeded. We could also just add a dummy column in the example, but I suppose this is the best way to ensure that the fix works in 1.6, and also makes the test a little more robust. LGTM, thanks.